### PR TITLE
workflow: fix hanging issue for error workflow

### PR DESCRIPTION
**Pull request by masco**

### CODE REVIEW

**Comment by masco-enovance** [comment]
rerun skydive-functional-hw-tests

**Comment by masco-enovance** [comment]
rerun skydive-functional-hw-tests

**Comment by masco-enovance** [comment]
rerun skydive-functional-hw-tests

**Comment by masco-enovance** [comment]
run skydive-functional-hw-tests

File: js/runtime.go [hunk 1]
```diff
@@ -404,6 +404,9 @@ func (r *Runtime) ExecFunction(source string, params ...interface{}) (v otto.Val
 func (r *Runtime) ExecPromise(source string, params ...interface{}) (v otto.Value, err error) {
 	var done chan otto.Value
 	r.Do(func(vm *otto.Otto) { done, err = r.CallPromise(source, params...) })
+	if err != nil {
+		return r.MakeCustomError("exec_error", err.Error()), err
+	}
 	v = <-done
 	return v, err
 }
```
Comment (hunchback) [inline js/runtime.go]: I would reduce the scope setting `v` to within the internal func:

```
// ExecPromise executes a promise and return its result
func (r *Runtime) ExecPromise(source string, params ...interface{}) (v otto.Value, err error) {
	var done chan otto.Value
	r.Do(func(vm *otto.Otto) { 
            done, err = r.CallPromise(source, params...)
	    if err == nil {
                v <-done
            } else {
		v = r.MakeCustomError("exec_error", err.Error())
	    }
        });
	return v, err
}

Comment (hunchback) [reply js/runtime.go]: I notice that return err within the value is not consistent throughout `runtime.go` so my suggestion is to handle this consistently throughout - if the design decision in-case-of error is to return the error within the value field - then this has to be followed through consistently ... (possibly in a separate PR).

```
// Exec queues the execution of some JavaScript code
func (r *Runtime) Exec(code string) (v otto.Value, err error) {
	r.Do(func(vm *otto.Otto) { v, err = vm.Run(code) })
	return v, err
}

// ExecFunction queues a CallFunction method
func (r *Runtime) ExecFunction(source string, params ...interface{}) (v otto.Value, err error) {
	r.Do(func(vm *otto.Otto) { v, err = r.CallFunction(source, params...) })
	return v, err
}

// ExecPromise executes a promise and return its result
func (r *Runtime) ExecPromise(source string, params ...interface{}) (v otto.Value, err error) {
	var done chan otto.Value
	r.Do(func(vm *otto.Otto) { done, err = r.CallPromise(source, params...) })
	if err != nil {
		return r.MakeCustomError("exec_error", err.Error()), err
	}
	v = <-done
	return v, err
}
```

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by masco-enovance** [comment]
run skydive-functional-tests-backend-orientdb

**Pull request closed without merging**